### PR TITLE
Add tree stories

### PR DESCRIPTION
**Pull request by edlu77**
#660

### CODE REVIEW

**Review by LibbyUX — COMMENTED** [review]
Hi Edward, this is looking great. Thanks for getting this so close!

One question:
- Is it possible to situate the collapsed parent icon in the specified state?

Thanks so much!

image

**Comment by edlu77** [comment]
@LibbyUX I've changed the icon, it should be using the correct one now.

Question about keyboard navigation, do we still want the same behavior as in the FTU Explorer?

**Comment by LibbyUX** [comment]
Hi @edlu77, yes please to the same keyboard tab functionality with the focus visible state.

Thanks so much for knocking this out!

**Comment by edlu77** [comment]
@LibbyUX Made some updates and the keyboard navigation should be working now. Let me know if there are any other changes needed!

**Review by LibbyUX — COMMENTED** [review]
Good morning, @edlu77! Thanks for the work on this! The tab looks good.

I just noticed there is a blue outline after pressing, probably because of the specified focus state variant. 

Is it possible to distinguish focus vs focus visible on this component? If not, let's just remove the focus state and go with focus visible only. I will delete the focus variant from our Figma library, too. 

I designed this component thinking we would be able to have separate looks for focus and focus visible, after a conversation with @axdanbol. No worries if it isn't possible, just let me know! Thanks so much and happy Tuesday! 

72E1AD89-0DA3-44BF-9D88-1E345CE19688

File: libs/design-system/tree/src/lib/tree-styles/tree-styles.component.scss [hunk 1]
```diff
@@ -0,0 +1,62 @@
+mat-tree {
+  --mat-tree-node-text-font: 'Metropolis';
```
Comment (axdanbol) [inline libs/design-system/tree/src/lib/tree-styles/tree-styles.component.scss:2]: The font values should come from variables. Set the value to `var(--sys-label-medium-font)` and `var(--sys-label-medium-weight)` for the weight

File: libs/design-system/tree/src/lib/tree-styles/tree-styles.component.scss [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  --mat-tree-node-text-weight: 500;
+  --mat-tree-container-background-color: white;
```
Comment (axdanbol) [inline libs/design-system/tree/src/lib/tree-styles/tree-styles.component.scss:4]: Use `var(--sys-primary-container)`

File: libs/design-system/tree/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './lib/tree-size/tree-size.directive';
+export * from './lib/tree.component';
```
Comment (axdanbol) [inline libs/design-system/tree/src/index.ts:2]: Don't export the demo component

File: libs/design-system/tree/src/lib/tree.component.ts
Comment (axdanbol) [inline libs/design-system/tree/src/lib/tree.component.ts:1]: Please rename this file and component to `TreeDemoComponent` (and the html file)

File: libs/design-system/tree/src/lib/tree-size/tree-size.directive.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ */
+@Directive({
+  selector: '[hraTreeSize]',
+  standalone: true,
+  host: {
+    '[style.--mat-tree-node-text-size.rem]': 'fontSize()',
+    '[style.--mat-tree-node-min-height.rem]': 'nodeHeight()',
+    '[style.--mdc-icon-button-state-layer-size.rem]': 'nodeHeight()',
+    '[style.--mdc-icon-button-icon-size.rem]': 'iconSize()',
+    '[style.--node-margin.rem]': '0.5',
```
Comment (axdanbol) [inline libs/design-system/tree/src/lib/tree-size/tree-size.directive.ts:38]: Please move constants to the tree-styles scss file

**Review by axdanbol — CHANGES_REQUESTED** [review]
A few small tweaks

**Comment by edlu77** [comment]
@LibbyUX Yes, that is the focus state. I agree it's a bit redundant since we already have a selected state which usually overrides it. You can only see this on expandable nodes that are clicked (since they can't be selected). I can remove the outline to make it cleaner.

**Review by axdanbol — APPROVED** [review]

**Pull request merged**